### ISSUE TITLE
CLI to verify images

### ISSUE THREAD

**Issue by mudler** [rancher-sandbox/cOS-toolkit#182]
After we have identified and implemented all the pieces to verify images during the various phases, we should encompass the single pieces in a single plugin which can be enabled during the appropriate phases. In this way the verification process is transparent to the user (building a derivative) and the CI

**Additional context**
Follow up of #95 , see also #126 

Blocked by #175 

## Requirements

 - [ ] single cli that works to generate checksums during build by hooking into events
 - [x] single cli that works to check checksums during upgrade by hooking into events
 - [ ] single cli that verifies checksums by passing a docker image
 - [x] single cli that generates checksums manually with parameters
 - [x] single cli that verifies checksums manually with parameters

**Comment by mudler** [comment]
@Itxaka would it be possible to make it work also standalone? e.g.:

```$> cli docker_image```

**Comment by Itxaka** [comment]
@mudler sure, really easy, the current implementation of the event action does exactly that so I could piggyback onto that by faking an event or just extracting the meat of the download, extract, compare into a different pig and call that from both places.


Is this a requirement of the current card? If so, I will update the card to reflect that 👍

**Comment by mudler** [comment]
yup, would really be required, our concrete use case would go mostly along the line of manual docker images

**Comment by Itxaka** [comment]
I think this is outdated, we should create a new card with any requirements if this is still needed.
